client/components/edit/EditParentsButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { useEditorContext } from "../../context/EditorContext";
+import ObjectLoader from "./ObjectLoader";
+
+export interface ObjectStatusProps {
+    pid: string;
+}
+
+export const EditParentsButton = ({ pid }: ObjectStatusProps): React.ReactElement => {
+    const {
+        state: { objectDetailsStorage },
+        action: { setParentsModalActivePid, toggleParentsModal },
+    } = useEditorContext();
+    const loaded = Object.prototype.hasOwnProperty.call(objectDetailsStorage, pid);
```

Comment:
What's the purpose of this function? Can you just set a default value?

Reply:
The `<ObjectLoader>` component has a side effect of loading object details into storage... so this makes the component self-contained. You can put it anywhere, and it will load the data it needs to operate, without depending on some other higher-level component. Whether or not this is a good idea is a separate question, but that's the purpose of the load check.

If we want to keep this approach, we could probably simplify the function to `return loaded ? <button>...</button> : <ObjectLoader ... />` instead of what it is currently doing, and it would have the same effect.

Alternatively, we could eliminate all the ObjectLoader stuff from here, and do the checking at a higher-level component where the button is displayed.

Reply:
In closer review of the code, there's no compelling reason to incorporate ObjectLoader functionality into the EditParentsButton, so I've simplified it with commit b5f4eaf7c3ca05720e11f1d967be1c506ba67405. Thanks for catching that bit of over-engineering!